### PR TITLE
refactor: delete defaultprops

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor: delete defaultprops |
| 🇨🇳 Chinese | refactor: delete defaultprops |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/anchor/AnchorLink.tsx [hunk 1]
```diff
@@ -14,40 +14,45 @@ export interface AnchorLinkProps {
   className?: string;
 }
 
-class AnchorLink extends React.Component<AnchorLinkProps, any, AntAnchor> {
-  static defaultProps = {
-    href: '#',
-  };
-
+class AnchorLink extends React.Component<AnchorLinkProps, {}, AntAnchor> {
```
Comment (zombieJ) [inline components/anchor/AnchorLink.tsx:17]: 这个去掉反而不好，全打散了。感觉要先改成 FC 才行。

File: components/input/__tests__/textarea.test.tsx [hunk 2]
```diff
@@ -3,7 +3,14 @@ import type { ChangeEventHandler, TextareaHTMLAttributes } from 'react';
 import React, { useState } from 'react';
 import Input from '..';
 import focusTest from '../../../tests/shared/focusTest';
-import { fireEvent, waitFakeTimer, render, sleep, triggerResize, pureRender } from '../../../tests/utils';
+import {
```
Comment (zombieJ) [inline components/input/__tests__/textarea.test.tsx:6]: 没变化的文件不放到 PR 里~

Comment (li-jia-nan) [reply components/input/__tests__/textarea.test.tsx:6]: 这个是 prettier 自动格式化的，我一保存就会变，不知道之前为什么 lint 会校验通过……

File: components/anchor/AnchorLink.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/anchor/AnchorLink.tsx]: 我把 anchor 回滚掉，后面单独提一个pr

**Pull request closed without merging**